collectivo/extensions/shifts/server/schemas/001_schema_shifts.ts

```diff
@@ -0,0 +1,208 @@
+// This function creates an empty schema for version 0.0.1 of the shifts extension
+// A schema can be used to declaratively define the structure of the database
+import { ShiftUserType } from "../utils/ShiftUserType";
+import { ShiftLogType } from "../utils/ShiftLogType";
+
+const schema = initSchema("shifts", "0.0.1");
+
+export default schema;
+
+const shiftUserTypeDropdownChoices = [];
+
+for (const type of Object.values(ShiftUserType)) {
+  shiftUserTypeDropdownChoices.push({ text: type, value: type });
+}
+
+const shiftLogTypeDropdownChoices = [];
+
+for (const type of Object.values(ShiftLogType)) {
+  shiftLogTypeDropdownChoices.push({ text: type, value: type });
+}
+
+schema.collections = [
+  {
+    collection: "shifts_shifts",
+    schema: {
+      schema: "schema",
+      name: "schema",
+      comment: null,
+    },
+    meta: {},
+  },
+  {
+    collection: "shifts_slots",
+    schema: {
+      schema: "schema",
+      name: "schema",
+      comment: null,
+    },
+    meta: {},
+  },
+  {
+    collection: "shifts_skills",
+    schema: {
+      schema: "schema",
+      name: "schema",
+      comment: null,
+    },
+    meta: {},
+  },
+  {
+    collection: "shifts_assignments",
+    schema: {
+      schema: "schema",
+      name: "schema",
+      comment: null,
+    },
+    meta: {},
+  },
+  {
+    collection: "shifts_logs",
+    schema: {
+      schema: "schema",
+      name: "schema",
+      comment: null,
+    },
+    meta: {},
+  },
+];
+
+schema.fields = [
+  ...directusSystemFields("shifts_shifts"),
+  {
+    collection: "shifts_shifts",
+    field: "shifts_name",
+    type: "string",
+    schema: {},
+    meta: {},
+  },
+  {
+    collection: "shifts_shifts",
+    field: "shifts_from",
+    type: "dateTime",
+    schema: { is_nullable: false },
+    meta: {},
+  },
+  {
+    collection: "shifts_shifts",
+    field: "shifts_to",
+    type: "dateTime",
+    schema: { is_nullable: true },
+    meta: {},
+  },
+  {
+    collection: "shifts_shifts",
+    field: "shifts_duration",
+    type: "integer",
+    schema: { is_nullable: false },
+    meta: { note: "In minutes" },
+  },
+  {
+    collection: "shifts_shifts",
+    field: "shifts_repeats_every",
+    type: "integer",
+    schema: { is_nullable: true },
+    meta: { note: "In days" },
```

Comment:
i would add another field to make it an option whether to repeat each days/weeks/months - then the admin user could set it e.g. to "repeats every fourth week on monday", or every 1st monday of each month?

Reply:
That would make more of what RRule offers available, that'd be nice, I'll update.